src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,137 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final int forNullKeyIndex = 0;
+    private static final double LOAD_FACTOR = 0.75;
+    private int threshold;
+    private int capacity;
+    private MapNode<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        capacity = INITIAL_CAPACITY;
+        table = createTable(capacity);
+        threshold = (int) (capacity * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        MapNode<K, V> toPut = new MapNode<>(key, value);
+        MapNode<K, V> currentNode = table[calculateIndex(toPut)];
+        int index = calculateIndex(toPut);
+        if (currentNode == null) {
+            table[index] = toPut;
+            size++;
+        } else {
+            while (true) {
```

Comment:
```suggestion
            size++;
            return;
        } 
        while (true) {
```
nested, indented code is harder to read, in situations like this, it's better to return early and unwrap rest of the code, so it's easier to follow